src/main/java/io/leonis/algieba/spatial/UniformFlowPotentialField.java

```diff
@@ -0,0 +1,98 @@
+package io.leonis.algieba.spatial;
+
+import io.leonis.algieba.calculus.ParametricLineIntegral;
+import java.util.function.UnaryOperator;
+import lombok.Value;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.factory.Nd4j;
+
+/**
+ * The Class UniformFlowPotentialField.
+ *
+ * This class represents a uniform flow {@link PotentialField} as proposed in section 2.1 of
+ * <a href="https://www.ri.cmu.edu/pub_files/pub3/kim_jin_oh_1992_2/kim_jin_oh_1992_2.pdf">this paper</a>.
+ *
+ * @author Rimon Oz
+ */
+@Value
+public class UniformFlowPotentialField implements ParametricLineIntegral, PotentialField {
+  /**
+   * The angle at which the flow inside the {@link PotentialField} flows.
+   */
+  private final double angle;
+  /**
+   * The slope of the {@link PotentialField}.
+   */
+  private final double strength;
+  /**
+   * The flow multiplier.
+   */
+  private final INDArray multiplier;
+  private final INDArray origin;
+
+  /**
+   * Constructs a uniform flow {@link PotentialField}.
+   * @param origin   The origin of the {@link PotentialField}.
+   * @param angle    The angle at which the uniform flow flows.
+   * @param strength The slope of the {@link PotentialField}
+   */
+  public UniformFlowPotentialField(final INDArray origin, final double angle,
+      final double strength) {
+    this.origin = origin;
+    this.angle = angle;
+    this.strength = strength;
+    this.multiplier = Nd4j
```

Comment:
Does this need to be calculated when the object is instantiated; or are you just compensating for the lack of good memorization in java ? (options that come to mind: [lombok's lazy getter](https://projectlombok.org/features/GetterLazy) or [Map#computeIfAbsent](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#computeIfAbsent-K-java.util.function.Function-))

Reply:
it's computed on instantiation and cached it since it doesn't change

Reply:
Yeah i commented because i think that we should have little code in the constructor as possible; but the memoization techniques in java are infinitely more shitty than doing it in the constructor.